File: framework/test2/floating/single/floating.rb
# step [1]: implement solution function
# @param {Float} a
# @param {Float} b
# @return {Float}
def divide(a, b)
  a / b
end

if __FILE__ == $0
  require 'soda/unittest/work'
  ns = Soda::Unittest
  # step [2]: setup function/return/arguments
  work = ns::TestWork.create(method(:divide))
  # work = ns::TestWork.forStruct(CLASS)
  # work.validator = -> (e, r) { e is equal to r }
  # work.compareSerial = true
  puts work.run(ARGF.read)
end

File: test-case/floating/single/floating.php
<?php
require_once 'Soda/DS/index.php';
require_once 'Soda/Unittest/index.php';

use Soda\DS;
use Soda\Unittest\TestWork;
use Soda\Unittest\Utils;

class Solution {

    /**
     * @param Double $x
     * @param Double $y
     * @return Double
     */
    function divide($a, $b) {
        return $a / $b;
    }
}

$work = TestWork::create(new Solution(), 'divide', $argv[0]);
// $work->validator = fn($e, $r) => <boolean result>
// $work->compareSerial = true;
echo $work->run(Utils::fromStdin());
